libtransmission/rpcimpl.cc

```diff
@@ -1648,6 +1660,16 @@ static char const* torrentAdd(tr_session* session, tr_variant* args_in, tr_varia
         tr_ctorSetFilePriorities(ctor, std::data(files), std::size(files), TR_PRI_HIGH);
     }
 
+    if (tr_variantDictFindList(args_in, TR_KEY_labels, &l))
+    {
+        auto labels = tr_labels_t{};
+        char const* errmsg = makeLabels(&labels, l);
+        if (errmsg == nullptr)
+        {
+            tr_ctorSetLabels(ctor, std::move(labels));
+        }
```

Comment:
I might be misreading this but should `setLabels()` be used here, and should this block return errmsg if setting fails?

Reply:
Right, I didn't know how to handle this. `setLabels` doesn't take a `tr_ctor` arg so I figured it needed a separate function. But you're right, this error message is basically useless and silently fails to add labels. I don't see how error messages are returned on other optional arguments to `torrent-add`. Maybe I wasn't looking hard enough?

Reply:
I guess the return is the errmsg, but should it fail on optional arguments?